main.py

```diff
@@ -69,42 +103,50 @@ def shutdown(self):
         If you have some cleanup procedure, insert it here.
         """
 
-    @rest('/v2/evc/', methods=['GET'])
+    @rest("/v2/evc/", methods=["GET"])
     def list_circuits(self):
         """Endpoint to return circuits stored.
 
         If archived is set to True return all circuits, else only the ones
         not archived.
         """
-        log.debug('list_circuits /v2/evc')
-        archived = request.args.get('archived', False)
+        log.debug("list_circuits /v2/evc")
+        archived = request.args.get("archived", False)
         circuits = self.storehouse.get_data()
         if not circuits:
             return jsonify({}), 200
         if archived:
             return jsonify(circuits), 200
-        return jsonify({circuit_id: circuit
-                        for circuit_id, circuit in circuits.items()
-                        if not circuit.get('archived', False)}), 200
-
-    @rest('/v2/evc/<circuit_id>', methods=['GET'])
+        return (
+            jsonify(
+                {
+                    circuit_id: circuit
+                    for circuit_id, circuit in circuits.items()
+                    if not circuit.get("archived", False)
+                }
+            ),
+            200,
+        )
+
+    @rest("/v2/evc/<circuit_id>", methods=["GET"])
     def get_circuit(self, circuit_id):
         """Endpoint to return a circuit based on id."""
-        log.debug('get_circuit /v2/evc/%s', circuit_id)
+        log.debug("get_circuit /v2/evc/%s", circuit_id)
         circuits = self.storehouse.get_data()
 
         try:
             result = circuits[circuit_id]
         except KeyError:
-            result = f'circuit_id {circuit_id} not found'
-            log.debug('get_circuit result %s %s', result, 404)
-            raise NotFound(result)
-
+            result = f"circuit_id {circuit_id} not found"
+            log.debug("get_circuit result %s %s", result, 404)
+            raise BadRequest(
+                json.dumps({"httpStatus": 400, "message": result})
```

Comment:
The API server in `kytos` core application caputres all HTTP exceptions and generate a message including the HTTP return code.
https://github.com/kytos-ng/kytos/blob/master/kytos/core/api_server.py#L68-L82

So, here you just need to send a message, not the status. And also you don't need to dump to JSON.

Reply:
Hi @ajoaoff, this is the initial error running with the original code:main.py|181:12|Unreachable code (W0101, unreachable):184:-1
main.py:Consider explicitly re-raising using the 'from' keyword (W0707, raise-missing-from) [pylint]

:( 1 issue found.
Linter check failed. Fix the error(s) above and try again.
ERROR: InvocationError for command /home/ubuntu/kytos/mef_eline/.tox/py36/bin/python setup.py lint (exited with code 255) (exited with code 255)